docs/guides/typescript.md

```diff
@@ -71,6 +71,229 @@ Providing the context and events as generic parameters for the `createMachine()`
 - The event type (`TEvent`) ensures that only specified events (and built-in XState-specific ones) are used in transition configs. The provided event object shapes are also passed on to actions, guards, and services.
 - Events which you send to the machine will be strongly typed, offering you much more confidence in the payload shapes you'll be receiving.
 
+## The VSCode Extension <Badge text="4.26+" />
+
+Using our [VSCode extension](https://marketplace.visualstudio.com/items?itemName=mattpocock.xstate-vscode), you can automatically generate intelligent typings for XState.
+
+Here's how you can get started:
+
+1. Download and install the [VSCode extension](https://marketplace.visualstudio.com/items?itemName=mattpocock.xstate-vscode).
+2. Open a new file and create a typed model using `createModel`. [Want more info on models?](./models.md). For instance:
+
+```ts
+const model = createModel(
+  {
+    value: ''
+  },
+  {
+    events: {
+      FOO: (value: string) => ({ value }),
+      BAR: () => ({})
+    }
+  }
+);
+```
+
+3. Create a machine from the model:
```

Comment:
a note that we should stick to "named" implementations here would be highlight this as a requirement for this feature